src/code_oscar.jl

```diff
@@ -0,0 +1,566 @@
+Return an array of bitsets,
+containing at position i the bitset for the i-th entry of `orbits`,
+that is, `true` at the positions of those cones in `orbits[i]`
+that contain the point `point`.
+"""
+function compute_bit_list(orbits, point)
+    bitset_list = map(i->BitSet(), orbits)
+    for current_orbit_nr in 1:length(orbits)
+        current_orbit = orbits[current_orbit_nr]
+        for current_cone_nr in 1:length(current_orbit)
+            current_cone = current_orbit[current_cone_nr]
+            if Polymake.polytope.contains(current_cone, point)
+                push!(bitset_list[current_orbit_nr], current_cone_nr)
+            end
+        end
+    end
+    return bitset_list
+end;
+
+"""
+    cones_from_bitlist(cone_list, bit_list_tuple)
+
+> Return the array of all cones at `true` positions in the bitsets,
+> where `cone_list` is an array of arrays of cones,
+> and `bit_list_tuple` is an array of bitsets.
+"""
+function cones_from_bitlist(cone_list, bit_list_tuple)
+    return_list = Any[]
+    for i in 1:length(cone_list)
+        for j in bit_list_tuple[i]
+            push!(return_list, cone_list[i][j])
+        end
+    end
+    return return_list
+end;
+
+
+hash_to_cone(orbit_list, hash) = Polymake.polytope.intersection(
+    cones_from_bitlist(orbit_list, hash)...);
+
+
+"""
+    get_neighbor_hash(orbits, facet_point, inner_normal_vector)
+
+> Return the list of bitsets describing the cones adjacent to the point
+> `facet_point` in the direction of `inner_normal_vector`.
+"""
+function get_neighbor_hash(orbits, facet_point, inner_normal_vector)
+    lambda = 1024
+    facet_point_bl = compute_bit_list(orbits, facet_point)
+    while true
+        current_point = lambda * facet_point - inner_normal_vector
+## FIXME: compute only necessary part of BL
+        current_bl = compute_bit_list(orbits, current_point)
+        if all(i->issubset(current_bl[i], facet_point_bl[i]), 1:length(facet_point_bl))
+            return current_bl
+        end
+        lambda *= 2
+    end
+end;
+
+"""
+    orbit_cone_orbits_and_action(I::Oscar.MPolyIdeal{T}, Q::Array{Int,2}, G::Oscar.GAPGroup) where {T}
+
+Return a dictionary containing
+- `orbit_list`: the array of orbit cone orbits,
+- `homs`: the array of the corresponding GAP homomorphism objects from `G` to the induced permutation action on the orbits,
+- `Q`: the matrix `Q`.
+
+Here `I` is the ideal in question, `Q` is the Q-matrix,
+and `G` is a symmetry group of the problem.
+"""
+function orbit_cone_orbits_and_action(I::Oscar.MPolyIdeal{T}, Q::Array{Int,2}, G::Oscar.GAPGroup) where {T}
+    collector_cones = orbit_cones(I, Q, G)
+    hom = action_on_target(Q, G)
+    orbit_list = orbit_cone_orbits(collector_cones, hom; disjoint_orbits = true);
+    homs = action_on_orbit_cone_orbits(orbit_list, hom)
+
+    return Dict("orbit_list" => orbit_list,
+                "hom" => hom,
+                "homs" => homs,
+                "Q" => Q,
+                "G" => G)
```

Comment:
This should be either a named tuple, or a proper struct (if this data is meant to be used in lots of other places)

Reply:
Based on the comments I read below, perhaps that `struct` could be named `GITFan`...? :-)